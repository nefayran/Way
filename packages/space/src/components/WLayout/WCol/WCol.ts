import { computed, h, defineComponent } from "vue";

export default defineComponent({
  name: "WCol",

  props: {
    col: {
      type: [Boolean, String, Number],
      default: 1,
    },
    tag: {
      type: String,
      default: "div",
    },
  },

  setup(props, { slots }) {
    const classes = computed(() => {
      const classList: any[] = [];

      classList.push({
        [`col-${props.col}`]: props.col,
      });

      return classList;
    });

    return () =>
      h(
        props.tag,
        {
          class: ["w-col col", classes.value],
        },
        slots.default?.()
      );
  },
});
