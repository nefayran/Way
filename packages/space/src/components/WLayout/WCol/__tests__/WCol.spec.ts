import { mount } from "@vue/test-utils";
import WCol from "../WCol";

describe("WCol.ts", () => {
  test("should have an w-col class", () => {
    const wrapper = mount(WCol, {
      propsData: {
        col: "2",
      },
    });
    expect(wrapper.classes()).toContain("w-col");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
