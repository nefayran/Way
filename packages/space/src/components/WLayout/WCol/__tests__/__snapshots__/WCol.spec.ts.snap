// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WCol.ts should have an w-col class 1`] = `<div class="w-col col col-2"></div>`;
